query-algebrizer/src/clauses/pattern.rs

```diff
@@ -238,8 +238,28 @@ impl ConjoiningClauses {
                 if value_type.is_none() {
                     self.wheres.add_intersection(ColumnConstraint::HasType(col.clone(), typed_value_type));
                 }
+            },
+        }
 
+        // TODO pattern.tx -> like handling entity place 
+        // carefully read Datomic query doc to check for correct handling
+        match pattern.tx {
+            PatternNonValuePlace::Placeholder => (),
+            PatternNonValuePlace::Variable(ref v) => {
+                self.bind_column_to_var(schema, col.clone(), DatomsColumn::Tx, v.clone());
+            },
+            PatternNonValuePlace::Entid(entid) => {
+                self.constrain_column_to_entity(col.clone(), DatomsColumn::Tx, entid);
```

Comment:
TODO: we want to check whether the tx-id is within range for the database's `tx` partition. (That applies after ident lookup, too.)